core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -161,6 +168,146 @@ angular.module('oppia').directive('adminMiscTab', [
+            });
+            // This will catch and log any error that occurs internally in
+            // MathJax.
+            MathJax.Hub.Register.MessageHook(
+              'Math Processing Error', function(message) {
+                LoggerService.error(message[2]);
+                reject();
+              });
+          });
+        };
+
+        var latexMapping = {};
+        var expIdToLatexMapping = null;
+        var numberOfLatexStrings = 0;
+        ctrl.generateSvgs = async function() {
+          var countOfSvgsGenerated = 0;
+          for (var expId in expIdToLatexMapping) {
+            var latexStrings = expIdToLatexMapping[expId];
+            latexMapping[expId] = {};
+            for (var i = 0; i < latexStrings.length; i++) {
+              var svgFile = await convertLatexToSvgFile(latexStrings[i]);
+              countOfSvgsGenerated++;
+              LoggerService.info(
+                'generated ' + countOfSvgsGenerated.toString() + ' SVGs');
+              latexMapping[expId][latexStrings[i]] = svgFile;
+            }
+            if (numberOfLatexStrings === countOfSvgsGenerated) {
+              ctrl.setStatusMessage(
+                'SVGs generated for ' + countOfSvgsGenerated.toString() +
+                ' LaTeX strings .');
+              $rootScope.$apply();
+            }
+          }
+        };
+
+        // TODO(#10045): Remove this function once all the math-rich text
+        // components in explorations have a valid math SVG stored in the
+        // datastore.
+        ctrl.fetchAndGenerateSvgsForExplorations = function() {
+          $http.get(ADMIN_MATH_SVG_IMAGE_GENERATION_HANDLER).then(
+            function(response) {
+              var numberOfExplorationsFetched = 0;
+              expIdToLatexMapping = (
+                response.data.latex_strings_to_exp_id_mapping);
+              for (var expId in expIdToLatexMapping) {
+                numberOfExplorationsFetched++;
+                numberOfLatexStrings += Object.keys(
+                  expIdToLatexMapping[expId]).length;
+              }
+              ctrl.setStatusMessage(
+                numberOfLatexStrings.toString() +
+                ' LaTeX strings fetched from backend for ' +
+                numberOfExplorationsFetched.toString() + ' explorations.' +
+                ' Generating SVGs.....');
+              ctrl.generateSvgs();
+            });
+        };
+        // TODO(#10045): Remove this function once all the math-rich text
+        // components in explorations have a valid math SVG stored in the
+        // datastore.
+        ctrl.saveSvgsToBackend = function() {
+          AdminDataService.sendMathSvgsToBackendAsync(
+            latexMapping).then(
+            function(response) {
+              var numberOfExplorationsUpdated = (
+                response.number_of_explorations_updated);
+              ctrl.setStatusMessage(
+                'Successfully updated ' + numberOfExplorationsUpdated +
```

Comment:
Along with this message, should also display the number of explorations left to be converted?
that will be useful right?

Reply:
Yes if it's easy to retrieve, otherwise don't worry.

Reply:
done